File: sharedbike/src/main/java/com/example/sharedbike/entity/Bike.java
package com.example.sharedbike.entity;

import com.example.sharedbike.entity.Enum.Status;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.util.Date;

@Data
public class Bike {
    private int bikeid;
    private String brand;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private Date  releasedate;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private Date  lastusetime;
    private int warrantyPeriod;
    private float LocationX;
    private float LocationY;
    private Status status;

    // Getters and setters
}

File: bikesystemfrontend/src/types/bike.ts
/*
 * @Description: 
 * @Version: 
 * @Author: DZQ
 * @Date: 2024-06-14 15:05:58
 * @LastEditors: DZQ
 * @LastEditTime: 2024-06-15 03:22:18
 */
export interface Bike {
  bikeid: number;
  brand: string;
  status: string;
  releasedate: string;
  warrantyPeriod: number;
  lastusetime: string;
  locationX: number; 
  locationY: number;

  }

export type bikeData = Bike;